include/RAJA/index/ListSegment.hpp

```diff
@@ -223,6 +281,23 @@ class TypedListSegment
   //! accessor to retrieve the total number of elements in a TypedListSegment
   RAJA_HOST_DEVICE Index_type size() const { return m_size; }
 
+  //! Create a slice of this instance as a new instance but with no ownership
+  //! of the data it points to.
+  /*!
+   * \return A new instance spanning *begin() + begin to *begin() + begin +
+   * length
+   */
+  RAJA_HOST_DEVICE RAJA_INLINE TypedListSegment slice(Index_type begin,
+                                                      Index_type length) const
+  {
+    Index_type end = begin+length > m_size ? (m_size-begin) : length;
+#if !defined(__CUDA_ARCH__)
+    return TypedListSegment(&m_data[begin], end, Owned);
```

Comment:
Is `Owned` correct?

Reply:
I had initially tried Unowned but it did not work with CPU compilers. I think there might be a thread privatizing issue if its not owned.

Reply:
Semantically it's wrong because the slice is not supposed to be owning the data right.

Reply:
I see, I think there may be a bug in tiling. Currently tiling relies on the slice method to do bounds checking. I changed the data ownership to Unowned and made a note on the .hpp file. Furthermore, the unit test are testing basic mechanics.